src/content/marina/marinaBroker.ts

```diff
@@ -100,9 +102,27 @@ export default class MarinaBroker extends Broker {
     if (!this.hostnameEnabled(state)) throw new Error('User must authorize the current website');
   }
 
+  private async reloadCoins(store: BrokerProxyStore) {
+    const state = store.getState();
+    const network = selectNetwork(state);
+    const allAccountsIds = selectAllAccountsIDs(state);
+    const makeUpdateTaskForId = (id: AccountID) => updateTaskAction(id, network);
+    allAccountsIds.map(makeUpdateTaskForId).map((x: any) => store.dispatch(x));
+    // wait max 10 seconds for update to finish
+    let attempts = 10;
```

Comment:
not sure 10 seconds could be enough for "deep" wallets to complete: what would happen in that case?

the essence of this API is to "load" the latest state possible of the utxo from blockchain, no matter how much time it takes.

Reply:
Good question.

It would load the same, but the request (ex: getCoins()) could be returning from a previous utxo set.

Spent days trying to figure how to force wait for reloadCoins() before proceeding. With the help of @louisinger this was the possible solution. Not happy about this solution, but not sure on how to implement it any other way.

Reply:
I tried with
```
while (store.getState().wallet.updaterLoaders !== 0) {}
```
but this becomes an infinite loop, since the event loop is never freed to perform the requires updateTask and so updaterLoaders never gets to 0.

Reply:
I do think reloadCoins is just suboptimal solution, which is the reason we do background updates in the first place: no need to call calls in sync when needed, but just get from storage

Reply:
I think the way we implement loaders is limited for that case: currently, `updaterLoaders` is an integer and let us know if there are **one or more** updaters running. To wait for a specific task properly, I think it misses two pieces:
* identity each UPDATE_TASK by an unique ID 
* associated a loader to each updater ID in the marina redux state (replace `updaterLoaders`)

something like:
```js
const id = marina.reloadCoins()
while (selectUpdaterIsLoading(id)(marinaStore.getState())) {
   // wait....
}
// done 
```

Reply:
Agreed, toughts @bordalix ?

Reply:
Agree, 5ef5337 is going into that direction.